src/common/transformations/include/transformations/op_conversions/convert_reduce_to_pooling.hpp

```diff
@@ -72,7 +72,7 @@ ov::matcher_pass_callback ConvertReduceBase::convert_reduce_to_pooling() {
     return [&](ov::pass::pattern::Matcher& m) {
         auto reduce = std::dynamic_pointer_cast<T>(m.get_match_root());
 
-        if (!reduce || transformation_callback(reduce)) {
+        if (!reduce || transformation_callback(reduce) || ov::shape_size(reduce->input_value(0).get_shape()) == 0) {
```

Comment:
```suggestion
        if (!reduce || ov::shape_size(reduce->input_value(0).get_shape()) == 0 || transformation_callback(reduce)) {        
```
Detail: Check shape size first then call callback (could be more complex than shape check)

Reply:
I was thinking about that order, and in my opinion the callback should have priority, as it allows plugin to inject any custom logic, even for the empty tensor case.